File: AndroidProject/app/src/main/java/com/example/androidproject/interfaces/MainFragmentListener.java
package com.example.androidproject.interfaces;

import com.example.androidproject.enums.BottomNavEnum;
import com.example.androidproject.enums.MenuViewEnum;

import java.io.Serializable;

public interface MainFragmentListener extends Serializable {
    void hideMenuView(MenuViewEnum viewEnum);

    void showHomeIcons();

    void showBottomNav();

    void setBottomNavItem(BottomNavEnum bottomNavItem);

    void showSearchView();

    void showFilterView();

    void showAccIcons();

    void showMarketIcons();

    void showCreateFareIcons();

    void hideBottomNav();

    void showLoginIcons();

    void showDetailFragmentIcons();

    void setSearchListener(OnSearchListener listener);
}

File: AndroidProject/app/src/main/java/com/example/androidproject/interfaces/MainFragmentListener.kt
package com.example.androidproject.interfaces

import com.example.androidproject.enums.BottomNavEnum
import com.example.androidproject.enums.MenuViewEnum
import java.io.Serializable


interface MainFragmentListener : Serializable {
    fun hideMenuView(viewEnum: MenuViewEnum?)
    fun showHomeIcons()
    fun showBottomNav()
    fun setBottomNavItem(bottomNavItem: BottomNavEnum?)
    fun showSearchView()
    fun showFilterView()
    fun showAccIcons()
    fun showMarketIcons()
    fun showCreateFareIcons()
    fun hideBottomNav()
    fun showLoginIcons()
    fun showDetailFragmentIcons()
}